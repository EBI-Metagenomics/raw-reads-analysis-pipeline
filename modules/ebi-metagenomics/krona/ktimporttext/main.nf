// This krona/ktimporttext module is copied from the already-existing nf-core module (https://nf-co.re/modules/krona_ktimporttext, https://github.com/nf-core/modules/commit/8fc1d24c710ebe1d5de0f2447ec9439fd3d9d66a)
// This is because there are not currently any nf-core ways of adding modules from more than one nf-core repo
// One slight change compared to the original is I've added a stub to this module

process KRONA_KTIMPORTTEXT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/krona:2.8.1--pl5321hdfd78af_1':
        'biocontainers/krona:2.8.1--pl5321hdfd78af_1' }"

    input:
    tuple val(meta), path(report)

    output:
    tuple val(meta), path ('*.html'), emit: html
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    ktImportText  \\
        $args \\
        -o ${prefix}.html \\
        $report

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        krona: \$( echo \$(ktImportText 2>&1) | sed 's/^.*KronaTools //g; s/- ktImportText.*\$//g')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.html

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        krona: \$( echo \$(ktImportText 2>&1) | sed 's/^.*KronaTools //g; s/- ktImportText.*\$//g')
    END_VERSIONS
    """
}
