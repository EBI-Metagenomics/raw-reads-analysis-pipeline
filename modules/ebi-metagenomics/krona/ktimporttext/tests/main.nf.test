// This krona/ktimporttext unit test file is converted to nf-test from an existing pytest test at this commit: https://github.com/nf-core/modules/commit/5e34754d42cd2d5d248ca8673c0a53cdf5624905
// I have converted it because the standard way of writing nf-core unit tests now is to use nf-test
// It is otherwise exactly the same unit test

nextflow_process {

    name "Test Process KRONA_KTIMPORTTEXT"
    script "../main.nf"
    process "KRONA_KTIMPORTTEXT"

    tag "modules"
    tag "modules_nfcore"
    tag "krona"
    tag "krona/ktimporttext"

    test("test_krona_ktimporttext - multi") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/krona/ktimporttext.txt', checkIfExists: true),                       // krona default test file
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/metagenome/test_1.kraken2.report.txt', checkIfExists: true), // Kraken2 report file
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/krona/kaiju_out4krona.txt', checkIfExists: true)                     // Kaiju output 4 krona
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(process.out.get(0).get(0).get(1)).readLines()[0].contains("DOCTYPE html PUBLIC")).match("multi_contains_DOCTYPE_html_PUBLIC") }
            )
        }

    }

    test("test_krona_ktimporttext - single") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file('http://krona.sourceforge.net/examples/text.txt', checkIfExists: true) // krona default test file
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(process.out.get(0).get(0).get(1)).readLines()[0].contains("DOCTYPE html PUBLIC")).match("single_contains_DOCTYPE_html_PUBLIC") }
            )
        }

    }

}
