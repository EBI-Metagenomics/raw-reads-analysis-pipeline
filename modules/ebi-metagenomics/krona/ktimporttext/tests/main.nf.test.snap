{
    "multi_contains_DOCTYPE_html_PUBLIC": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-17T11:15:59.756328"
    },
    "single_contains_DOCTYPE_html_PUBLIC": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-17T11:16:15.282079"
    }
}